### ISSUE TITLE
How do taxcalc's difference functions handle reforms that change expanded income?

### ISSUE THREAD

**Issue by MaxGhenis** [open-source-economics/Tax-Calculator#1893]
Expanded income is the default "dimension" / x-axis for slicing effects of reforms. But this includes benefits, and UBI in reforms where it's present. This means that reforms involving these parameters alter expanded income.

How do taxcalc's difference functions handle these circumstances? It seems like there are two options:

1. Calculate impact on each expanded income bin under the baseline and reform separately, and report the difference. This is what I had thought the behavior was, but it would present problems.

2. Bin only on the baseline, and map tax units back to their baseline expanded income when calculating the reform effect. That difference tables report the total # tax units suggests that this is the behavior, since otherwise there could be different counts.

Given this potential confusion, it may be useful to have an income measure that's independent of benefits and UBI, as a more "fixed" x-axis. As it stands, expanded income is somewhere between an immutable measure (can still change under dynamic models, but assumed constant for static) and after-tax income.

**Comment by martinholmer** [comment]
@MaxGhenis,  The documentation on the Calculator `distribution_tables` method is pretty clear about this.
```
    def distribution_tables(self, calc,
                            groupby='weighted_deciles',
                            income_measure='expanded_income',
                            result_type='weighted_sum'):
        """
        Get results from self and calc, sort them based on groupby using
        income_measure, manipulate grouped statistics based on result_type,
        and return tables as a pair of Pandas dataframes.
        Note that the returned tables have consistent income groups (based
        on the self income_measure) even though the income_measure in self
        and the income_measure in calc are different.
```

**Comment by martinholmer** [comment]
@MaxGhenis said:
> Given this potential confusion, it may be useful to have an income measure that's independent of benefits and UBI, as a more "fixed" x-axis. As it stands, expanded income is somewhere between an immutable measure (can still change under dynamic models, but assumed constant for static) and after-tax income.

First of all, I don't see any "potential confusion".  But I guess you do.  
What income measure would you suggest as an alternative to baseline expanded income?

**Comment by MaxGhenis** [comment]
@martinholmer Thanks, I'd been looking at the [overall documentation](https://open-source-economics.github.io/Tax-Calculator/) and for [`difference_table`](https://github.com/open-source-economics/Tax-Calculator/blob/92d32ff2a0d9055e8c0fd2a72e970a8e881da5fa/taxcalc/calculate.py#L468) which is a bit less explicit. It sounds like difference tables then use approach 1, but this only answers part of my question, so let me be more specific:

* Does this mean that the count of "All tax units" in the table is the count under the baseline? For reforms that affect benefits or UBI, the count under the reform would be similar for quantiles, but could differ under the baseline.

* Are changes to tax liability and after-tax income calculated by summing over each bin and then the difference? It sounds like yes, in which case they could be affected by how the reform moves tax units between bins.

Hypothetical example: $10k UBI moves nearly all tax units out of the $0-10k expanded income bin. After-tax income for this bin could then actually *decrease*, since UBI for tax units who were in that group under the baseline is counted in higher income bins.

Let me know if my questions aren't clear and I can put together a quick notebook with an example that would demonstrate my concern more explicitly.

> What income measure would you suggest as an alternative to baseline expanded income?

Without knowing how other analysis groups approach this, a new measure defined as *expanded income minus benefits minus UBI* seems natural.

**Comment by martinholmer** [comment]
@MaxGhenis asked:
> Thanks, I'd been looking at the overall documentation and for difference_table which is a bit less explicit.

You're right about the `difference_table` docstring needing better documentation.  Here is what it says now in pending pull request #1890:
```
     def difference_table(self, calc,
                          groupby='weighted_deciles',
                          income_measure='expanded_income',
                          tax_to_diff='combined'):
         """
         Get results from self and calc, sort them based on groupby using
         income_measure, and return tax-difference table as a Pandas dataframe.
         This method leaves the Calculator objects unchanged.
         Note that the returned tables have consistent income groups (based
         on the self income_measure) even though the baseline income_measure
         in self and the income_measure in calc are different.
         Note that filing units are put into groupby categories using the
         specified income_measure in the baseline (self) situation.
         Also, note that some subgroups may contain filing units with negative
         or zero baseline (self) income.
```
> Does this mean that the count of "All tax units" in the table is the count under the baseline? 

Yes.

> For reforms that affect benefits or UBI, the count under the reform would be similar for quantiles, but could differ when binned.

I don't understand the above statement.  Could you try posing your question in different words?

> Are changes to tax liability and after-tax income calculated by summing over each bin and then the difference? It sounds like yes, in which case they could be affected by how the reform moves tax units between bins.

Changes in "tax liability and after-tax income" are computed for each filing unit.  Then the bin/quantile change is computed as the sum of the changes for each filing unit in the bin/quantile.

> Hypothetical example: $10k UBI moves nearly all tax units out of the $0-10k expanded income bin. After-tax income for this bin could then actually decrease, since UBI for tax units who were in that group under the baseline is counted in higher income bins.

As you can see from my answers to your prior questions, this would never happen in Tax-Calculator tables or graphs.  The bins/quantiles are always defined using baseline (that is, pre-reform) income.  We have been doing things this way since the beginning of the project and we are under the impression that this is how other tax analysis groups do things.

**Comment by MaxGhenis** [comment]
> Changes in "tax liability and after-tax income" are computed for each filing unit. Then the bin/quantile change is computed as the sum of the changes for each filing unit in the bin/quantile.

Thanks, this is the key, and maps to option 2 in my [original comment](https://github.com/open-source-economics/Tax-Calculator/issues/1893#issue-299041591).

Here's a [notebook](http://nbviewer.jupyter.org/github/MaxGhenis/taxcalc-notebooks/blob/master/exploratory/ubi_10k_baseline_binning.ipynb) illustrating the impact. This makes sense for the difference tables, but is less intuitive for the distribution tables where I expected that they were computed independently. Out of curiosity, is this because the distribution table is used for calculating the difference table?

My original question is answered, though I'm curious what you think about a different expanded income measure. Something like this would essentially be the pre-government analog of after-tax income, the post-government conclusion of taxcalc. I'm not sure if there were other government programs before UBI and benefits that went into expanded income, but at this point it's no longer pre-government. If you think it's worth a discussion we can continue in a separate issue.

**Comment by martinholmer** [comment]
@MaxGhenis asked:
>> Changes in "tax liability and after-tax income" are computed for each filing unit. Then the bin/quantile change is computed as the sum of the changes for each filing unit in the bin/quantile.
>
> Thanks, this is the key, and maps to option 2 in my original comment.
>
> ... This makes sense for the difference tables, but is less intuitive for the distribution tables where I expected that they were computed independently (without reading docs). Out of curiosity, is this because the distribution table is used for calculating the difference table?

No, they are computed independently.  The reason the post-reform distribution table uses pre-reform bins/quantiles is that we expect users will difference the two (pre-reform and post-reform) distribution tables and be puzzled if the difference in a decile's tax liability is not the same as corresponding statistic in the difference table for that same reform.

**Comment by martinholmer** [comment]
@MaxGhenis asked:
> I'm curious what you think about a different expanded income measure. Something like this would essentially be the pre-government analog of after-tax income, the post-government conclusion of taxcalc. I'm not sure if expanded income included other government programs before UBI and benefits, but at this point it's no longer pre-government. If you think it's worth a discussion we can continue in a separate issue.

In response to your subquestion:
> I'm not sure if expanded income included other government programs before UBI and benefits

Yes, expanded income includes benefits from several "government programs" including unemployment insurance benefits and social security benefits.  A substantial fraction of social security benefits are included in AGI.  I'm not sure how we could sensibly define a "pre-government" income statistic.  Would we exclude all wages and salaries paid by federal, state, and local governments?  My understanding is that other tax analysis groups tend to use a broad definition of income.

**Comment by MaxGhenis** [comment]
Thanks for the context @martinholmer. Do analysts ever slice by after-tax income, and quantiles of those? The raw number might be less interpretable since people think of their own paycheck, but at least for the quantiles I imagine it maps not too far off, and removes confusion about what programs are included in expanded income.

**Comment by feenberg** [comment]
On Wed, 21 Feb 2018, Martin Holmer wrote:

> 
> @MaxGhenis asked:
>
>             Changes in "tax liability and after-tax income" are computed for each filing
>             unit. Then the bin/quantile change is computed as the sum of the changes for
>             each filing unit in the bin/quantile.
>
>       Thanks, this is the key, and maps to option 2 in my original comment.
>
>       ... This makes sense for the difference tables, but is less intuitive for the distribution
>       tables where I expected that they were computed independently (without reading docs). Out
>       of curiosity, is this because the distribution table is used for calculating the
>       difference table?
> 
> No, they are computed independently. The reason the post-reform distribution table uses pre-reform
> bins/quantiles is that we expect users will difference the two (pre-reform and post-reform)
> distribution tables and be puzzled if the difference in a decile's tax liability is not the same as
> corresponding statistic in the difference table for that same reform.

If one looks at a distribution table arranged by pre-tax income, and the 
column showing the change in tax, one can generally tell if the change in 
tax is progressive or not. If one is comparing the distribution of income 
under two different tax regimes by comparing after-tax incomes, and wishes 
to know if one is more progressive than the other, then each table needs 
to be arranged using the after-tax income of that regime and needs to show 
the after-tax income.

I believe the tables I have seen from TB are in the first category. If we 
have any that show after-tax incomes, then they should be tabbed by 
after-tax income.

In short, if we show a change in tax or income, tab by initial income. If 
we show total tax or income from both regimes, tab by each regime 
separately. If we show a change and a total, tab by the regime that 
creates the total.

dan

**Comment by feenberg** [comment]
On Wed, 21 Feb 2018, Martin Holmer wrote:

> 
> @MaxGhenis asked:
>
>       I'm curious what you think about a different expanded income measure. Something like this
>       would essentially be the pre-government analog of after-tax income, the post-government
>       conclusion of taxcalc. I'm not sure if expanded income included other government programs
>       before UBI and benefits, but at this point it's no longer pre-government. If you think
>       it's worth a discussion we can continue in a separate issue.
> 
> In response to your subquestion:
>
>       I'm not sure if expanded income included other government programs before UBI and benefits
> 
> Yes, expanded income includes benefits from several "government programs" including unemployment
> insurance benefits and social security benefits. A substantial fraction of social security benefits
> are included in AGI. I'm not sure how we could sensibly define a "pre-government" income statistic.
> Would we exclude all wages and salaries paid by federal, state, and local governments? My
> understanding is that other tax analysis groups tend to use a broad definition of income.

I think he means "before taxes and transfers", which is a commonly used 
metric.

dan

**Comment by feenberg** [comment]
The last paragraph of mine below is wrong and should be ignored. I believe 
the rest is OK.

On Thu, 22 Feb 2018, Daniel Feenberg wrote:

>
>
> On Wed, 21 Feb 2018, Martin Holmer wrote:
>
>> 
>> @MaxGhenis asked:
>>
>>             Changes in "tax liability and after-tax income" are computed 
>> for each filing
>>             unit. Then the bin/quantile change is computed as the sum of 
>> the changes for
>>             each filing unit in the bin/quantile.
>>
>>       Thanks, this is the key, and maps to option 2 in my original comment.
>>
>>       ... This makes sense for the difference tables, but is less intuitive 
>> for the distribution
>>       tables where I expected that they were computed independently 
>> (without reading docs). Out
>>       of curiosity, is this because the distribution table is used for 
>> calculating the
>>       difference table?
>> 
>> No, they are computed independently. The reason the post-reform 
>> distribution table uses pre-reform
>> bins/quantiles is that we expect users will difference the two (pre-reform 
>> and post-reform)
>> distribution tables and be puzzled if the difference in a decile's tax 
>> liability is not the same as
>> corresponding statistic in the difference table for that same reform.
>
> If one looks at a distribution table arranged by pre-tax income, and the 
> column showing the change in tax, one can generally tell if the change in tax 
> is progressive or not. If one is comparing the distribution of income under 
> two different tax regimes by comparing after-tax incomes, and wishes to know 
> if one is more progressive than the other, then each table needs to be 
> arranged using the after-tax income of that regime and needs to show the 
> after-tax income.
>

This next paragraph is not right. Sorry:

**Comment by MaxGhenis** [comment]
@feenberg I'm not totally clear on your comments (do you want to edit your earlier one?), but it seems like there are few potential combinations of x and y. Which boxes are considered valid?

| Metric | Income before taxes and transfers* | Expanded income minus UBI minus benefits** | Expanded income | After-tax income (includes UBI and benefits) |
| --- | --- | --- | --- | --- |
| % change in after-tax income | | | Supported in taxcalc |
| % change in tax liability | | | Supported in taxcalc |

\* May not be available in taxcalc.
\** taxcalc's closest approximation of income before taxes and transfers?

Also from @feenberg's two uses of the term "progressive" it sounds like this can refer to either (1) income before taxes and transfers, or (2) income after taxes and transfers (based on the baseline scenario), is that right?

It seems to me that the main advantage of slicing by income before taxes and transfers is that these are equal between baseline and reform for tax units in static analysis. Without this advantage, unless it deviates significantly from other analysis groups, baseline after-tax income seems simpler by not introducing a third variable.
